[OSPO Book] Include initial content for chapter 2 and 3 (#299)
todogroup/ospology: ospo-book/chapters/02-chapter.md

# Chapter 2: Understanding the Value of Open Source Program Offices

- [Introduction](#introduction)
    - Reasons Behind Starting an OSPO
    - Reasons Behind Sustaining an OSPO
- [Assessing Value of Open Source Activity](#assessing-value-of-open-source-activity) - `✅ Assessment`
- [Antipatterns](#antipatterns) - `🚫 OSPO Antipatterns`
- [Resources](#resources) - `📚 Continue Here`

# Introduction

> From end-user companies and software companies to universities and public administrations, an organization's relationship with open source and the value it can get from an OSPO depend on the type of organization it is. However, we can still identify some critical problems that arise when getting involved in open source on an organizational level, which motivates the creation and continuity of an OSPO.

Reviewer: Having to deal with open source is the reality for every organizations which uses software in some way. So I would see the argumentation not so much in challenges which arise when getting involved with open source, but necessary steps which have to be taken no matter what. To deal with software in a responsible and professional way requires to deal with open source. Doing this requires some organizational structure and that can an OSPO be.

> From end-user companies and software companies to universities and public administrations, an organization's relationship with open source and the value it can get from an OSPO depend on the type of organization it is. However, we can still identify some critical problems that arise when getting involved in open source on an organizational level, which motivates the creation and continuity of an OSPO.

Reviewer 2: I would say that the relationship with open source and the value it can get from an OSPO not only depend on the type of organisatie, but differs for each organisation. 

```suggestion
From end-user companies and software companies to universities and public administrations, an organization's relationship with open source and the value it can get from an OSPO is different per organization. However, we can still identify some critical problems that arise when getting involved in open source on an organizational level, which motivates the creation and continuity of an OSPO.
```

* **The tension between the need to ship product features and the need to contribute back to open source:** Open source contributions may take a back seat when dealing with multiple day-to-day tasks.

* **Procurement processes with never-ending steps:** Open source is a dynamic ecosystem and contributions to it should occur as smoothly and naturally as possible. The long procurement processes faced in highly regulated environments, such as finance companies and governments, create a barrier to open source contribution and engagement.

* **A lack of consciousness about organizational responsibility:** Due to the way open source was taught in the past, engineering-based tools, or even the engineering jargon used, the concept of open source may not be taken seriously in other areas of the organization involved in decision-making processes, management, or policy making.

To fully overcome these and other problems when engaging in open source, organizations need to be equipped to manage open source operations effectively on both cultural and practical levels.

> To understand the value of an OSPO, the reader needs to get familiar to the reasons behind (1) starting an OSPO and (2) sustaining an OSPO in the long term

Reviewer: I'm not sure the reasons to start an OSPO are much different from sustaining an OSPO. I would more look at it as reasons for **having** an OSPO.

Side note: The "program" in OSPO might mislead some organizations into thinking that this is a temporary activity which has an end, when the program is finished. That's why I would focus on the responsibility of an OSPO to manage open source as the ongoing activity which might evolve over time to become more sophisticated, but is certainly not a time-limited activity with a projected end.

![ospowork](https://user-images.githubusercontent.com/43671777/232471414-8ea7c899-a13c-47de-bafe-836830525d02.png)

*[Source:OSPOs, key lever for open source sustainability](https://speakerdeck.com/anajsana/ospos-a-key-lever-for-open-source-sustainability)* 

## The reasons behind STARTING an OSPO

The [business value of the OSPO](https://project.linuxfoundation.org/hubfs/LF%20Research/TODO%20Group%20-%20The%20Business%20Value%20of%20the%20OSPO%20-%20Report.pdf?hsLang=en) explains some of the reasons shared by Open Source leaders across different industries and organization sizes.
* Building standardized processes around open source
* Learn how to approach the open source community

> * Embracing the Sustainability of Open Source Projects

Author: Thanks for pointing this out @CsatariGergely  We'll keep it in mind 🙂 Please note that there will be a proofread work once we finalize the first version of the book, and this is one of the things that will be fixed. But the sooner we notice this and fix it, the less work it will take later 👍

> * Mitigating Security Risks
> * Embracing the Sustainability of Open Source Projects
> * Managing Compliance 
> * Building standardized processes around open source
> * Learn how to approach the open source community
> * Expanding access to open knowledge
> * Improving development velocity

Reviewer 3: I think we need to reorder these since the chapter mentioned contributing to open source and procedures first on the above. I'm worried that `Mitigating Security Risks` might confuse readers when we list this first. 
```suggestion
* Building standardized processes around open source
* Learn how to approach the open source community
* Embracing the Sustainability of Open Source Projects
* Managing Compliance 
* Expanding access to open knowledge
* Improving development velocity
* Mitigating Security Risks
```

## The reasons behind SUSTAINING an OSPO

Stopping the work of an OSPO could have significant negative impacts on those organizations that use open source at any level, including loss of open source expertise, increased security risks, reduced community engagement, and damage to reputation.

> An OSPO needs to be an ongoing initiative within an organization in order to evolve its culture and open-source knowledge, helping the organization to contribute to and build more secure open-source software, as well as improving the sustainability of open-source projects. 

Reviewer 2: I would add that OSPO also helps in using open source software. In my experience this is one of bigest responsibilities of an OSPO. 

```suggestion
An OSPO needs to be an ongoing initiative within an organization in order to evolve its culture and open-source knowledge, helping the organization to use, to contribute to and build more secure open-source software, as well as improving the sustainability of open-source projects. 
```

* **Acts as a Facilitator:** The OSPO also plays a sort of translation role between
Organization’s teams and decision makers' interests regarding open source and the needs from teh open source community. They also help organizations navigate the cultural, process, and tool changes required to engage with the open source community effectively and in a healthy way.

* **Acts as an Advocate:** OSPOs can promote the use and/or contribution of open source and best practices across different organizational units. This can help organizations realize the benefits of open source as well as engaging people to contribute to open source projects or start new ones

* **Acts as an Environmentalist:** OSPOs can help organizations support and sustain open source projects in the long term by addressing issues such as security, maintenance, and project health. This can help ensure that open source projects remain healthy in the long term and continue to benefit the wider community.

> 

Reviewer 2: I would add a fifth Role:

```suggestion
**Acts as an Gatekeeper:** OSPOs can help to enfore OS policies and strengthen OS governance. This can help organisatie to ensure compliance and mitigate OS security risks.
```

Reviewer 4: Looks good, but "organisatie" should be changed to "organisations"

> **Acts as an Gatekeeper:** OSPOs can help to enforce OS policies and strengthen OS governance. This can help organizations to ensure compliance and mitigate OS security risks.

Reviewer 2: The formatting of the fifth role was not inline with the rest. A suggestion to align it:

```suggestion

* **Acts as an Gatekeeper:** OSPOs can help to enforce OS policies and strengthen OS governance. This can help organizations to ensure compliance and mitigate OS security risks.
```

* Improved Compliance: An OSPO helps public sector organizations to ensure that their open source operations are compliant with relevant laws and regulations, including data privacy laws, procurement regulations, and transparency requirements. This helps organizations to avoid costly legal and regulatory challenges, and to maintain their reputation as responsible public sector organizations.

* Increased Collaboration: An OSPO helps public sector organizations to foster collaboration between different departments and with external stakeholders, including other public sector organizations, open source communities, and civil society organizations. This increased collaboration helps organizations to access a wider pool of talent and resources, and to develop better open source solutions.

* Better Resource Allocation: An OSPO helps public sector organizations to allocate resources more effectively, ensuring that open source operations are well-supported and that key initiatives are given the resources they need to succeed. This helps organizations to maximize the benefits of open source technology, and to drive innovation and growth.

> * Improved Service Delivery: An OSPO helps public sector organizations to improve the delivery of public services, by enabling them to adopt innovative and cost-effective technologies, and to collaborate with external stakeholders to develop better solutions. This helps organizations to provide better services to citizens, and to meet the changing needs of their communities.

Reviewer 4: I think there are too many repetitions of "An OSPO helps public sector organizations" in here.

Author: improved!

In Chapter 1 we already introduced how open source has become increasingly significant for modern organizations. By extending the concept of "open" to encompass open research, design, and access, we can identify additional benefits that these practices bring to organizations. This broader view of openness is gaining traction in academic and public sectors, where the *open work* term is starting to be used among participants that engage in OSPO practitioner communities.

# Assessing Value of Open Source Activity

`✅ Assessment`

Despite an organizations might be aware of the general problems, responsibility and benefits that contributing to open source provides, identifying specific key motivators to move people to take action (create activity) and prioritize open source is a tough task. In this section, we will assess a methodology to communicate the value of contributing to open source and going beyond, which the OSPO can develop and use when working with the different teams that engages with open source. (Source: [ospo-book mailing list discussion](https://lists.todogroup.org/g/ospo-book-project/message/18))

> * Step one: Get familiar with the consumer-participation-contribution-leadership open source involvement model. [This report by Dr. Ibrahim H](https://8112310.fs1.hubspotusercontent-na1.net/hubfs/8112310/LF%20Research/LF%20Research%20Guide%20to%20Enterprise%20Open%20Source.pdf) explains the different stages (from consumer to leader involvement)

Reviewer 5: What about the Maturity Model from https://www.linuxfoundation.org/research/the-evolution-of-the-open-source-program-office-ospo ?

> * Step one: Get familiar with the consumer-participation-contribution-leadership open source involvement model. [This report by Dr. Ibrahim H](https://8112310.fs1.hubspotusercontent-na1.net/hubfs/8112310/LF%20Research/LF%20Research%20Guide%20to%20Enterprise%20Open%20Source.pdf) explains the different stages (from consumer to leader involvement)

Reviewer 5: What about the Maturity Model from https://www.linuxfoundation.org/research/the-evolution-of-the-open-source-program-office-ospo ?

Reviewer 5: sorry, clicked approve on the review by accident.  don't think I can undo that

Author: No worries @robmoffat! 

re: ospo model. I think that model is useful for the OSPO team to understand the maturity of its program and be able to communicate to other teams (or even internally) when advancing in their objectives. However, the Open Source Involvement model is a previous step IMHO, to map open source efforts on an organizational level first, to start educating the other teams on what does use, participation, contribution, and leadership means, and the kind of activity they are doing and aiming to do in the future. Once this is clear, OSPO teams can take a look at the OSPO model if that helps them to prioritize tasks and better define their roadmap across the years 🙂

* Step three: With the output from these 1:1 conversations, define concrete motivators and map them into areas within the organization where open source brings value
* Step four: make a second division that categorizes each of these motivators on the different stages within the open source involvement model

This is an example of the value of activity from participation to leadership stage that an OSPO from a corporation can elaborate on by following these steps:

<img width="942" alt="activityparticipationcategorization" src="https://user-images.githubusercontent.com/43671777/232468402-bb4a4d49-a6d6-4c74-8d21-88c6be3c2c13.png">

# Antipatterns

`🚫 OSPO Antipatterns`

Common misunderstandings that should be corrected as soon as possible

* **Characterize OSPOs as a purely top-down initiative:** For instance, one pushed by management on an unwilling team of engineers
* **OSPO = Sales Profit or Marketing Tool:** Assume that an OSPO will act as a lead generator.
* **Believe that an OSPO provides an immediate solution** to the open source problems the organization faces

> * **See OSPOs as an added value and not as direct support for the core organization’s areas and functions:** As cited in the 2020 State of the Software Supply Chain Report from Sonatype, [80-90% of modern applications consist of open source software](https://www.sonatype.com/state-of-the-software-supply-chain/open-source-dependency-management-trends-and-recommendations). if your organization uses open source in their internal processes and/or products, is very likely that the OSPO enable and maintain risk mitigation.

Reviewer 4: ```suggestion
* **See OSPOs as an added value and not as direct support for the core organization’s areas and functions:** As cited in the 2020 State of the Software Supply Chain Report from Sonatype, [80-90% of modern applications consist of open source software](https://www.sonatype.com/state-of-the-software-supply-chain/open-source-dependency-management-trends-and-recommendations). If your organization uses open source in their internal processes and/or products, is very likely that the OSPO enable and maintain risk mitigation.
```

> 

Reviewer 2: An fifth antipattern could be: 

```suggestion
**OSPO that have pure a technical focus and forget about Open Source culture:**  Working on code together means a lot of things: transparency, diversity and cooperation.
```
